### PR TITLE
Mocha chai sinon

### PR DESCRIPTION
**Pull request by gareththegeek**
Yeah baby!

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+                "ts-node/register",
+                "${relativeFile}",
+                "--timeout=0"
+            ],
+            "cwd": "${workspaceRoot}",
+            "protocol": "inspector",
+            "internalConsoleOptions": "openOnSessionStart"
+        },
+        {
+            "name": "All Tests",
```
Comment (dougajmcdonald) [inline .vscode/launch.json]: I like the two launch settings

File: README.md [hunk 2]
```diff
@@ -50,9 +50,7 @@ Beyond the main API, we envisage access to the Core object being provided via ev
 
 Corewar is extensively unit tested, run `npm test` to execute the tests and see the coverage.
 
-The tests run via `jasmine-ts` which uses `ts-node` under the hood, so as a result aren't easy to debug (by easy, read 'possible') :(
-
-You may need `jasmine-cli` installed globally if you don't have it already
+The tests run via `ts-mocha` which uses `ts-node` under the hood, so as a result aren't easy to debug (by easy, read 'possible') :(
```
Comment (dougajmcdonald) [inline README.md]: Attention to detail!

File: package.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
   "description": "HTML5 & javascript implementation of the classic game [corewar](https://en.wikipedia.org/wiki/Core_War)",
   "main": "./dist/index.js",
   "scripts": {
-    "test": "nyc jasmine-ts",
+    "test": "nyc mocha",
```
Comment (dougajmcdonald) [inline package.json]: Sic

File: package.json [hunk 4]
```diff
@@ -41,12 +36,15 @@
   },
   "homepage": "https://github.com/gareththegeek/corewar#readme",
   "devDependencies": {
-    "@types/jasmine": "^2.6.3",
-    "jasmine": "^2.8.0",
-    "jasmine-core": "^2.8.0",
-    "jasmine-ts": "^0.2.1",
+    "@types/mocha": "^2.2.44",
```
Comment (dougajmcdonald) [inline package.json]: nice

File: parser/tests/ForPassTests.ts [hunk 5]
```diff
@@ -8,14 +10,14 @@ import { MessageType } from "../interface/IMessage";
 import { TestHelper } from "./TestHelper";
 "use strict";
 
-describe("ForPass",() => {
+describe("ForPass", () => {
 
-    it("Inserts for instruction into token stream the requested number of times",() => {
+    it("Inserts for instruction into token stream the requested number of times", () => {
```
Comment (dougajmcdonald) [inline parser/tests/ForPassTests.ts]: Looks important! :P

File: simulator/tests/toEqualInstruction.d.ts [hunk 6]
```diff
@@ -1,5 +0,0 @@
-declare namespace jasmine {
```
Comment (dougajmcdonald) [inline simulator/tests/toEqualInstruction.d.ts]: Fuck this file basically!

Comment (dougajmcdonald) [reply simulator/tests/toEqualInstruction.d.ts]: All that effort and we just kill it...I'm almost sad...

File: simulator/tests/SimulatorTests.ts [hunk 7]
```diff
@@ -212,42 +216,42 @@ describe("Simulator",() => {
...
+        expect(core.initialise).to.have.been.calledWith(Defaults);
     });
 
     it("stores warriors returned from the loader in the state passed to the fetch method",() => {
 
         var warriors = [];
         var loadedWarriors = [];
 
-        (<jasmine.Spy>loader.load).and.returnValue(loadedWarriors);
+        (<sinon.stub>loader.load).returns(loadedWarriors);
```
Comment (dougajmcdonald) [inline simulator/tests/SimulatorTests.ts]: The syntax changes are interesting, some I like some I don't, this one I prefer

File: README.md (hunk 2 above)
Comment (gareththegeek) [reply README.md]: Did a global search for jasmine :)

File: simulator/tests/toEqualInstruction.d.ts (hunk 6 above)
Comment (gareththegeek) [reply simulator/tests/toEqualInstruction.d.ts]: Haha, yeah :/

**Pull request merged**